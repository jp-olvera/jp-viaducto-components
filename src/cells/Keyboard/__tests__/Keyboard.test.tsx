import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';

import { Keyboard } from '..';

describe('Keyboard', () => {
  test('onDown', () => {
    const onDown = jest.fn();
    const { container } = render(
      <div>
        <Keyboard onDown={onDown}>
          <span>content</span>
        </Keyboard>
      </div>
    );

    const element = screen.getByText('content');

    fireEvent.keyDown(element, { keyCode: 40 });
    fireEvent.keyDown(element, { which: 40 });
    fireEvent.keyDown(element, { which: 0 });

    expect(onDown).toHaveBeenCalled();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('onKeyDown', () => {
    const onDown = jest.fn();
    const onKeyDown = jest.fn();
    const { container } = render(
      <div>
        <Keyboard onDown={onDown} onKeyDown={onKeyDown}>
          <span>content</span>
        </Keyboard>
      </div>
    );

    const element = screen.getByText('content');

    fireEvent.keyDown(element, { keyCode: 40 });

    expect(onDown).toBeCalled();
    expect(onKeyDown).toBeCalled();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('change onKeyDown', () => {
    const firstOnKeyDown = jest.fn();
    const secondOnKeyDown = jest.fn();

    const { container, getByText, rerender } = render(
      <Keyboard target='document' onKeyDown={firstOnKeyDown}>
        <span>content</span>
      </Keyboard>
    );
    expect(container.firstChild).toMatchSnapshot();

    fireEvent(
      getByText('content'),
      new KeyboardEvent('keydown', { bubbles: true, cancelable: true })
    );
    expect(firstOnKeyDown).toBeCalled();

    rerender(
      <Keyboard target='document' onKeyDown={secondOnKeyDown}>
        <span>content</span>
      </Keyboard>
    );
    expect(container.firstChild).toMatchSnapshot();

    fireEvent(
      getByText('content'),
      new KeyboardEvent('keydown', { bubbles: true, cancelable: true })
    );
    expect(secondOnKeyDown).toBeCalled();
  });
});
