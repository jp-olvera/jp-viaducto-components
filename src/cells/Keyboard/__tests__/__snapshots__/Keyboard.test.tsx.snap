// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard change onKeyDown 1`] = `
<span>
  hi
</span>
`;

exports[`Keyboard change onKeyDown 2`] = `
<span>
  hi
</span>
`;

exports[`Keyboard onDown 1`] = `
<div>
  <span>
    hi
  </span>
</div>
`;

exports[`Keyboard onKeyDown 1`] = `
<div>
  <span>
    hi
  </span>
</div>
`;
